examples/basic/basic.c

```diff
@@ -36,7 +35,10 @@ async_print_impl(struct future_context *ctx, struct future_notifier *notifier)
 	if (notifier) notifier->notifier_used = FUTURE_NOTIFIER_NONE;
 
 	struct async_print_data *data = future_context_get_data(ctx);
-	printf("async print: %p\n", data->value);
+	struct async_print_output *output = future_context_get_output(ctx);
+
+	int ret = printf("async print: %p\n", data->value);
```

Comment:
printf, on success, returns the number of printed characters.

maybe error_code should be `ret < 0 ? -ret : 0` ?